pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -129,17 +135,31 @@ type PullSource struct {
 // PullSourcesFromReference returns a slice of PullSource's based on the passed
 // reference.
 func (r *Registry) PullSourcesFromReference(ref reference.Named) ([]PullSource, error) {
+	if r.DigestOnly {
```

Comment:
Not sure if that's necessary here since it should be rejected when _loading_ the configuration files.